device/samples/pnp/pnpTemperatureController.js

```diff
@@ -97,30 +98,41 @@ const updateComponentReportedProperties = (deviceTwin, patch, componentName) =>
   });
 };
 
-const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin) => {
+const helperAttachHandlerForDesiredPropertyPatches = (deviceTwin, componentNames) => {
   deviceTwin.on('properties.desired', (delta) => {
-    if (delta[thermostat1ComponentName] || delta[thermostat2ComponentName] || delta[deviceInfoComponentName]) {
+    console.log('Received an update for device with value: ' + JSON.stringify(delta));
+    Object.entries(delta).forEach(([key, values]) => {
```

Comment:
> Object.entries(delta).forEach(([key, values]) => { [](start = 3, length = 51)

Consider the case of the application being out of date with the model.   So that this device only knows about components a,b, and c.  However, suppose that there is a new component d.

Consider further that a changes occur inside a, b, and c, and a writeable property of d.  so d.foo is changed.

The code will to reports for a, b, c in the block starting at line 106.  When it gets to d, the code will fall through to the block starting at 122.  It appears that it will report a,b,c, and d all over again.

Reply:
my assumption that user must know about the existence of all its components